spec/h2_connection_spec.lua

```diff
@@ -157,6 +159,40 @@ describe("http2 connection", function()
 			assert.truthy(cq:empty())
 		end)
 	end)
+	describe("priority on missing stream", function()
+		it("sets default priority for streams with missing parent", function()
+			local cq = cqueues.new()
+			local c, s = cs.pair()
+			cq:wrap(function()
+				c = h2_connection.new(c, "client")
+				local client_stream = c:new_stream()
+				local req_headers = new_headers()
+				req_headers:append(":method", "GET")
+				req_headers:append(":scheme", "http")
+				req_headers:append(":path", "/")
+				assert(client_stream:write_headers(req_headers, false))
```

Comment:
Yes. but there are some things that can go wrong with that.
I've got some stuff (locally) that I use to try and find leaks of streams/connections.

Otherwise, in general I encourage use of `stream:shutdown()` to release resources as soon as possible. Relying on `__gc` for this isn't great, as lots of people don't realise they're still hanging on to a reference.